src/erc7496/erc7496.cairo

```diff
@@ -0,0 +1,155 @@
+// @title DynamicTraits
+
+// @dev Implementation of [ERC-7496](https://eips.ethereum.org/EIPS/eip-7496) Dynamic Traits.
+
+// Requirements:
+// - Overwrite `setTrait` with access role restriction.
+// - Expose a function for `setTraitMetadataURI` with access role restriction if desired.
+#[starknet::component]
+mod ERC7496Component {
+    use openzeppelin::introspection::src5::SRC5Component::InternalTrait as SRC5InternalTrait;
+    use openzeppelin::introspection::src5::SRC5Component::SRC5;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use cairo_erc_7496::erc7496::interface::{IERC7496_ID, IERC7496};
+
+    #[storage]
+    struct Storage {
+        /// @dev A mapping of token ID to a mapping of trait key to trait value.
+        ERC7496_traits: LegacyMap<(u256, felt252), felt252>,
+        /// @dev An offchain string URI that points to a JSON file containing trait metadata.
+        ERC7498_trait_metadata_uri: ByteArray,
+    }
+
+    #[event]
+    #[derive(Drop, PartialEq, starknet::Event)]
+    enum Event {
+        TraitUpdated: TraitUpdated,
+        TraitMetadataURIUpdated: TraitMetadataURIUpdated
+    }
```

Comment:
given the official eip  : 
https://eips.ethereum.org/EIPS/eip-7496

you might need to also define `TraitUpdatedRange` , `TraitUpdatedRangeUniformValue` , `TraitUpdatedList` and `TraitUpdatedListUniformValue` events no?

Reply:
`Event` enum needs to be `pub` to be usable in the preset with edition 2023_11, better to add it now